src/js/components/AppListComponent.jsx

```diff
@@ -70,13 +70,19 @@ var AppListComponent = React.createClass({
     var sortKey = state.sortKey;
     var props = this.props;
 
-    return lazy(state.apps)
+    var appsSequence = lazy(state.apps);
+
+    if (props.filterText != null && props.filterText != "") {
+      appsSequence = appsSequence
+        .filter(function (app) {
+          return app.id.indexOf(props.filterText) !== -1;
```

Comment:
I find the lines in the example misleading

```
Lazy(["foo", "bar", "baz"]).indexOf("bar")   // => 1
Lazy([1, 2, 3]).indexOf(4)                   // => -1
Lazy([1, 2, 3]).map(reciprocal).indexOf(0.5) // => 1
``` 

To me that reads: this also works when not chaining. It suggests you would have an improvement even when using this outside of a chain (as in the first two lines).